extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,327 @@
+    style.setStyles(this.domElements_.CONTAINER, {
+      fontSize: '50px',
+      fontVariant: 'normal',
+      left: '-999px',
+      lineHeight: 'normal',
+      margin: 0,
+      padding: 0,
+      position: 'absolute',
+      top: '-999px',
+      visibility: 'hidden'
+    });
+    DEFAULT_FONTS_.forEach(font => {
+      let defaultFontElement = this.document_.createElement('div');
+      this.domElements_.DEFAULT_FONT_ELEMENTS.push(defaultFontElement);
+      defaultFontElement.textContent = TEST_STRING_;
+      style.setStyles(defaultFontElement, {
+        fontFamily: font,
+        margin: 0,
+        padding: 0,
+        whiteSpace: 'nowrap'
+      });
+      containerElement.appendChild(defaultFontElement);
+    });
+    // Adding custom font family to the element to trigger load.
+    // The loading will begin after the container has been appended to the body.
+    let customFontElement = this.domElements_.CUSTOM_FONT_ELEMENT =
+        this.document_.createElement('div');
+    style.setStyles(customFontElement, {
+      fontFamily: this.customFontFamily_,
+      margin: 0,
+      padding: 0,
+      whiteSpace: 'nowrap'
+    });
+    customFontElement.textContent = TEST_STRING_;
+    containerElement.appendChild(customFontElement);
+    this.document_.body.appendChild(containerElement);
+  }
+
+
+  /**
+   * Compare dimensions between elements styled with default fonts and custom
+   * font.
+   * @returns {boolean} Returns true if the dimensions are noticeably different
+   * else returns false.
+   * @private
+   */
+  compareMeasurements_() {
+    let hasWidthChanged = false;
+    let hasHeightChanged = false;
+    this.domElements_.DEFAULT_FONT_ELEMENTS.some(defaultElement => {
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetWidth -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetWidth) >
+            TOLERANCE_);
+      hasWidthChanged = (
+          Math.abs(
+              defaultElement./*REVIEW*/offsetHeight -
+              this.domElements_.CUSTOM_FONT_ELEMENT./*REVIEW*/offsetHeight) >
+            TOLERANCE_);
+      if (hasWidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if (hasWidthChanged || hasHeightChanged) {
+      return true;
+    }
+    return false;
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadSuccess_() {
+    timer.cancel(this.timeoutHandle_);
+    var addClassName = this.element.getAttribute('on-load-add-class');
+    var removeClassName =
+        this.element.getAttribute('on-load-remove-class');
+    if (addClassName) {
+      this.documentElement_.classList.add(addClassName);
+    }
+    if (removeClassName) {
+      this.documentElement_.classList.remove(removeClassName);
+      this.document_.body.classList.remove(removeClassName);
+    }
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadError_() {
+    var addClassName = this.element.getAttribute('on-error-add-class');
```

Comment:
You can refactor `onFontLoadSuccess` and `onFontLoadError` by extracting a method that would take two attribute names and do the rest by reading class names and applying them correctly.